DEVDOCS-5037: [new] GraphQL Cart and Checkout (#1953)
bigcommerce/dev-docs: docs/api-docs/storefront/graphql/graphql_carts_and_checkout.mdx

# GraphQL Carts and Checkout

> BigCommerce's GraphQL Storefront API now supports Cart and Checkout objects within the graph. This Cart and Checkout support offers similar benefits as our REST APIs, but having Cart and Checkout in GraphQL alongside other data makes it easier for building storefront functionality.

Reviewer: ```suggestion
BigCommerce's GraphQL Storefront API supports Cart and Checkout objects within the graph. This Cart and Checkout support offers similar benefits as our REST APIs, but having Cart and Checkout in GraphQL alongside other data makes it easier for building storefront functionality.
```

> The addition of GraphQL now means we have all the critical GraphQL capabilities necessary to power an end-to-end shopper purchase funnel in GraphQL, inclusive of:

Reviewer: ```suggestion
The addition of GraphQL now means we have all the critical GraphQL capabilities necessary to power an end-to-end shopper purchase funnel in GraphQL, including the following features:
```

* Obtaining the cost of shipping and selecting a shipping method
* Converting a cart into an order (completing checkout)

## Limitations

* The GraphQL API will give you the appropriate parameters for a request to our Payments API to choose a payment instrument. However, we have intentionally kept the transmission of payment data out of the GraphQL API for PCI reasons.
* Gift Certificate is now a payment instrument (i.e., a payment method similar to a credit card). The Payments API will need to apply the gift certificate to the checkout.
* This GraphQL Storefront Carts & Checkout API does not support file upload product options, cart metafields, BOPIS (Buy Online Pick Up In Store), and Buy Now cart where a shopper can check out a product on the Product Details Page, skipping the cart page.

## Tokens

Tokens need to be generated under the [Storefront API Tokens](/docs/storefront-auth/tokens#create-a-token) and [Storefront API Customer Impersonation Tokens](docs/storefront-auth/tokens/customer-impersonation-token) scopes. See [Creating a Token](/api-docs/storefront/graphql/graphql-api-overview#creating-a-token) for more information.

<Callout type="info">

> OAuth resource scopes are not applicable to Storefront API. Those scopes are only relevant to Admin/Management APIs (e.g. v3 APIs where you can edit admin data).

Reviewer: ```suggestion
OAuth resource scopes are not applicable to Storefront API. Those scopes are only relevant to Admin/Management APIs (e.g. V3 APIs where you can edit admin data).
```

</Callout>

## Example GraphQL queries and mutations

### Create a cart using a simple product

<Tabs items={['Request', 'Response']}>
  <Tab>

  ```graphql filename="Example mutation: Create a cart (simple)" showLineNumbers copy 
  # Creates a new cart, adding a simple product.

>   # This will fail if you already have a cart.

Reviewer: ```suggestion
  # This will fail if you already have a cart.
```
You might want to specify that it fails in the GraphQL playground, but in Postman, I think you might be able to make multiple carts and get the carts with the cart's entity ID
